File: backend/src/main/java/org/example/backend/dto/UserResultDTO.java
package org.example.backend.dto;

public record UserResultDTO(
        int userPossibleElectricityGeneration,
        int userAmountOfPossibleSavings,
        double userAmortisationTime,
        double userLifetimeYieldKwh,
        double userCo2SavingsKgPerYear,
        double userSelfConsumptionRate,
        double userAutarkyRate,
        double userDailyYield,
        double userDailySavings,
        double userHomeofficeCoverageRate,
        double userDailyEBikeRangeKm,
        double userDailyECarRangeKm
) {}

File: frontend/src/dto/UserResultDTO.ts
export interface UserResultDTO {
    userPossibleElectricityGeneration: number;
    userAmountOfPossibleSavings: number;
    userAmortisationTime: number;
    userLifetimeYieldKwh: number;
    userCo2SavingsKgPerYear: number;
    userSelfConsumptionRate: number;
    userAutarkyRate: number;
    userDailyYield: number;
    userDailySavings: number;
    userDailyEBikeRangeKm: number;
    userDailyECarRangeKm: number;
    userHomeofficeCoverageRate: number; // 0–1
}